account-compression/sdk/tests/accountCompression.test.ts

```diff
@@ -0,0 +1,684 @@
+      const replaceLeafIx = createReplaceIx(
+        payer,
+        splCMTKeypair.publicKey,
+        offChainTree.root,
+        previousLeaf,
+        newLeaf,
+        index,
+        proof
+      );
+      try {
+        await execute(provider, [replaceLeafIx], [payer]);
+        assert(false, "Replace should have failed to verify");
+      } catch { }
+
+      const splCMT = await getConcurrentMerkleTree(connection, splCMTKeypair.publicKey);
+      const onChainRoot = getCMTCurrentRoot(splCMT);
+
+      assert(
+        Buffer.from(onChainRoot).equals(offChainTree.root),
+        "Updated on chain root matches root of updated off chain tree"
+      );
+    });
+    it("Replace that leaf", async () => {
+      const previousLeaf = offChainTree.leaves[0].node;
+      const newLeaf = crypto.randomBytes(32);
+      const index = 0;
+
+      const replaceLeafIx = createReplaceIx(
+        payer,
+        splCMTKeypair.publicKey,
+        offChainTree.root,
+        previousLeaf,
+        newLeaf,
+        index,
+        getProofOfLeaf(offChainTree, index, false, -1).map((treeNode) => {
+          return treeNode.node;
+        })
+      );
+      assert(
+        replaceLeafIx.keys.length == 3 + MAX_DEPTH,
+        `Failed to create proof for ${MAX_DEPTH}`
+      );
+
+      await execute(provider, [replaceLeafIx], [payer]);
+
+      updateTree(offChainTree, newLeaf, index);
+
+      const splCMT = await getConcurrentMerkleTree(connection, splCMTKeypair.publicKey);
+      const onChainRoot = getCMTCurrentRoot(splCMT);
+
+      assert(
+        Buffer.from(onChainRoot).equals(offChainTree.root),
+        "Updated on chain root matches root of updated off chain tree"
+      );
+    });
+
+    it("Replace that leaf with a minimal proof", async () => {
+      const previousLeaf = offChainTree.leaves[0].node;
+      const newLeaf = crypto.randomBytes(32);
+      const index = 0;
+
+      const replaceLeafIx = createReplaceIx(
+        payer,
+        splCMTKeypair.publicKey,
+        offChainTree.root,
+        previousLeaf,
+        newLeaf,
+        index,
+        getProofOfLeaf(offChainTree, index, true, 1).map((treeNode) => {
+          return treeNode.node;
+        })
+      );
+      assert(
+        replaceLeafIx.keys.length == 3 + 1,
+        "Failed to minimize proof to expected size of 1"
+      );
+      await execute(provider, [replaceLeafIx], [payer]);
+
+      updateTree(offChainTree, newLeaf, index);
+
+      const splCMT = await getConcurrentMerkleTree(connection, splCMTKeypair.publicKey);
+      const onChainRoot = getCMTCurrentRoot(splCMT);
+
+      assert(
+        Buffer.from(onChainRoot).equals(offChainTree.root),
+        "Updated on chain root matches root of updated off chain tree"
+      );
+    });
+  });
+
+  describe("Examples tranferring appendAuthority", () => {
+    const authority = Keypair.generate();
+    const randomSigner = Keypair.generate();
+    describe("Examples transferring authority", () => {
```

Comment:
Is this extra level of indirection needed?

Reply:
Added a test case that shows failure of a fake signer trying to replace a leaf with a valid proof, but without the correct authority. I think that should help with the clarity of this test suite.

What do you mean by `extra level of indirection` ?

Reply:
Sorry, that wasn't clear -- on line 361, there's `describe("Examples tranferring appendAuthority", () => {` and then on 364 there's the nested `describe("Examples transferring authority", () => {`, which confused me a bit.  It seems like you only need one of those `describe`s

Reply:
Shoot, great catch again! Thank you Jon!!